apps/sms/js/thread_list_ui.js

```diff
@@ -393,17 +394,24 @@ var ThreadListUI = {
     }
   },
 
-  markReadUnread: function thlui_markReadUnread(selected, isRead) {
-    selected.forEach((id) => {
+  markReadUnreadUI: function thlui_markReadUnreadUI(selected, isRead) {
+    function filterIds(id) {
       var thread  = Threads.get(id);
       var markable = thread && (!thread.hasDrafts || isRead);
 
       if (markable) {
         thread.unreadCount = isRead ? 0 : 1;
-        this.mark(thread.id, isRead ? 'read' : 'unread');
-
-        MessageManager.markThreadRead(thread.id, isRead);
+        ThreadListUI.mark(thread.id, isRead ? 'read' : 'unread');
       }
+      return markable;
+    }
+
+    ThreadListUI.markReadUnread(selected.filter(filterIds), isRead);
```

Comment:
please use `this` instead of `ThreadListUI`, it helps understanding that we call something on the same object.